### PR TITLE
Use multiprocessing to process + fix bug with gitignore flag

### PR DESCRIPTION
**Pull request by inverse**
## Proposed change

- Process files using pool of workers to speed up the process (file collection is still an issue though on large projects)
- Fix bug with gitignore flag being set and path not including `.gitignore` file

In a future PR the logic for building the files to parse could also be improved as I noticed this was also a bottleneck on some large projects.

## How to test the change

You should see all cores being used during scan + speed improvements

## Checklist

-   [ ] Tests have been added to verify that the new code works (if possible)
-   [ ] Documentation has been updated to reflect changes

### CODE REVIEW

File: tell_me_your_secrets/__main__.py [hunk 1]
```diff
@@ -127,15 +130,20 @@ def match(self, file_path: str, file_content: str) -> MatchResult:
...
+                 user_filters: Optional[list] = None):
+        self.start_time = time.time()
         self.search_path = search_path
         self.use_gitignore = use_gitignore
         if use_gitignore:
             gitignore_file = os.path.join(search_path, '.gitignore')
             if os.path.exists(gitignore_file):
                 module_logger.debug(f'Using gitignore file: {gitignore_file}')
                 self.gitignore_matcher = parse_gitignore(gitignore_file)
+            else:
```
Comment (valayDave) [inline tell_me_your_secrets/__main__.py:144]: shouldn't it be in the same level of indent with `if use_gitignore`? asking based on the property you have set based on else statement.

Comment (inverse) [reply tell_me_your_secrets/__main__.py:144]: The logic could probably be simplified further but the logic which uses this failed when the matcher was not set. This means that it is set to false and thus not used.

**Comment by inverse** [comment]
@valayDave anything else to add?

**Review by valayDave — APPROVED** [review]

File: test/test_main.py [hunk 2]
```diff
@@ -1,35 +0,0 @@
...
-
-    def test_run_signatures_matched(self):
-        signature_recognizer = SignatureRecognizer({}, '.', False)
-        signature_recognizer.signatures.append(MockSignature(True, 'matched-yada'))
-
-        result = signature_recognizer.run_signatures('file/with/issues', 'dodgy-content')
-        self.assertEquals('Mock Signature', result[0])
-        self.assertEquals('file', result[1])
-
-    def test_run_signatures_whitelisted(self):
```
[the remark this answers is not in the document]
Comment (inverse) [reply test/test_main.py:28]: It was moved and modified, I dont know why git didn't persist this :(

**Pull request merged**